cylc/flow/task_job_mgr.py

```diff
@@ -1000,16 +1000,17 @@ def _simulation_submit_task_jobs(self, itasks, workflow):
             itask.waiting_on_job_prep = False
             itask.submit_num += 1
             self._set_retry_timers(itask)
-            itask.platform = {'name': 'SIMULATION'}
+            itask.platform['name'] = 'SIMULATION'
```

Comment:
We don't want the platform config in it's entirety replaced by `{'name': 'SIMULATION'}`. When we come around to retry with that config we have no info about the host &c to use. This is a bug!

Reply:
I think this was correct before.

Sure this is modifying the `localhost` platform that each new `TaskProxy` object is (incorrectly) assigned when it is initiated?

Reply:
It does however, need the platform default for `submission retry delays` added to this "platform" to prevent a failure in `_set_retry_timers`  on the retry.